### PR TITLE
Add support for years in MooseUnit

### PR DESCRIPTION
**Pull request by GiudGiud**
refs #27009

### CODE REVIEW

File: framework/src/utils/Units.C [hunk 1]
```diff
@@ -193,7 +193,8 @@ const std::vector<std::pair<std::string, MooseUnits>> MooseUnits::_unit_table =
        {MooseUnits::BaseUnit::SECOND, -2}}}},                        // bar (unit of pressure)
     {"h", {60 * 60, 0, {{MooseUnits::BaseUnit::SECOND, 1}}}},        // hour
     {"day", {60 * 60 * 24, 0, {{MooseUnits::BaseUnit::SECOND, 1}}}}, // day
-    {"l", {1e-3, 0, {{MooseUnits::BaseUnit::METER, 3}}}},            // liter
+    {"a", {60 * 60 * 24 * 365.2422, 0, {{MooseUnits::BaseUnit::SECOND, 1}}}}, // year (a for annum)
```
Comment (dschwen) [inline framework/src/utils/Units.C:196]: No. `a` unit is 365.25 days.

**Review by dschwen — CHANGES_REQUESTED** [review]
Fix number of days.

**Comment by dschwen** [comment]
Well I suppose the difference is that between a Julian calendar year (365.25) and a Gregorian calendar year (365.2425). In astronomy the Julian year is more commonly used.

File: framework/src/utils/Units.C (hunk 1 above)
Comment (GiudGiud) [reply framework/src/utils/Units.C:196]: probably should have checked wikipedia first

**Review by dschwen — CHANGES_REQUESTED** [review]
Could you please update the table in [Units.md](https://github.com/idaholab/moose/blob/next/framework/doc/content/source/utils/Units.md).

File: framework/doc/content/source/utils/Units.md [hunk 2]
```diff
@@ -59,6 +59,7 @@ The following units are currently supported
 | `bar` | $10^5\, \text{kg}\cdot \text{m}^{-1}\cdot \text{s}^{-2}$ | Bar | pressure, stress |
 | `h` | $3600\, \text{s}$ | hour | time |
 | `day` | $86400\, \text{s}$ | day | time (`d` would create an ambiguity between centi-day and candela) |
+| `year` | $365.25 \text{day}$ | annum | time (Julian calendar year, `a` would create an ambiguity between Peta-year and pascal) |
```
Comment (GiudGiud) [inline framework/doc/content/source/utils/Units.md:62]: @dschwen I did

Comment (GiudGiud) [reply framework/doc/content/source/utils/Units.md]: Added a space so it looks better

**Review by dschwen — APPROVED** [review]

**Pull request merged**